File: 2006-count-number-of-pairs-with-absolute-difference-k/2006-count-number-of-pairs-with-absolute-difference-k.cpp
class Solution {
public:
    int countKDifference(vector<int>& arr, int k) {
        // |nums[i] - nums[j]| == k 
        // nums[i] - nums[j] = k  || -nums[i] + nums[j] = k
        // nums[j] = nums[i] - k  || nums[i] + k ;
        // x +  y = k
        int cnt = 0 ;
        unordered_map<int,int> mp ;
        for(int i = 0 ; i < arr.size() ; i ++)
        {
            if(mp.find(arr[i] +k) != mp.end())
            {
                cnt += mp[arr[i] + k] ;
            }
            if(mp.find(arr[i] - k) != mp.end())
            {
                cnt += mp[arr[i] - k] ;
            }
            mp[arr[i]] ++;
        }
        return cnt ;
    }
};

File: 2006-count-number-of-pairs-with-absolute-difference-k/2006-count-number-of-pairs-with-absolute-difference-k.java
class Solution {
    public int countKDifference(int[] arr, int k) {
        int cnt = 0 ;
        HashMap<Integer , Integer > mp = new HashMap<>();
        for(int i = 0 ; i < arr.length ; i ++)
        {
            if(mp.containsKey(arr[i] + k))
            {
                cnt += mp.get(arr[i] + k) ;
            }
            if(mp.containsKey(arr[i] - k))
            {
                cnt += mp.get(arr[i] - k) ;
            }
            mp.put(arr[i] , mp.getOrDefault(arr[i] , 0) + 1) ;
        }
        return cnt ;
    }
}